### PR TITLE
vo_gpu_next: fix OSD rendering of screenshots

### LINKED ISSUE #9923: vo_gpu-next: incorrect subtitle's presentation in default screenshot mode
### Important Information

Provide following Information:
- mpv version
 https://github.com/mpv-player/mpv/commit/b1fb4b783bd604ffb763e88a7cee2ad9d9c2c209
- Windows Version
 win11 release
- Source of the mpv binary
 same with shinchiro's toolchain
- If known which version of mpv introduced the problem
 probably https://github.com/mpv-player/mpv/pull/9870
- Possible screenshot or video of visual glitches
 see below

### Reproduction steps

1. play the video with external subtitles
2. take a screenshot

### Expected behavior

At least the subtitle should be stay at the middle of bottom.
(created by ctrl+s)
mpv-shot0002

### Actual behavior

(created by s)
mpv-shot0001

### Log file

N/A

### Sample files

Any external text subtitles.

### LINKED ISSUE #9928: Subtitles don't move to the visible portion of the video in vo=gpu-next
### Important Information

Provide following Information:
- mpv version: https://github.com/mpv-player/mpv/commit/b1fb4b783bd604ffb763e88a7cee2ad9d9c2c209
- Windows Version: Windows 10 21H1 19043.1526
- Source of the mpv binary: shinchiro's build
- If known which version of mpv introduced the problem: probably #9870 
- Possible screenshot or video of visual glitches

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps

1.  Play a non 16:9 aspect video (in this case it's 1920x816)
2.  Take a screenshot

### Expected behavior
image
Subtitles are moved to the visible portion of the video (taken with screenshot window in vo=gpu and blend-subtitles=yes)
### Actual behavior
image
Subtitles are not there
image
This is the image taken with screenshot window in vo=gpu-next. Subtitles are shown on the the black bars but are not visible during playback
### Log file
http://0x0.st/oK78.txt

### PR DESCRIPTION
**Pull request by haasn**
One downside of this approach is that it bypasses the mixer cache, but
while this is not ideal for performance reasons, the status quo is also
simply broken so I'd rather have a slower implementation that works than
a faster implementation that does not.

And as it turns out, updating the OSD state and invalidating the mixer
cache correctly is sufficiently nontrivial to do in a clean way, so I'd
rather have this code that I can be reasonably certain does the right
thing.

Fixes #9923 as discussed. Also fixes #9928.

### CODE REVIEW

File: video/out/vo_gpu_next.c [hunk 1]
```diff
@@ -1109,8 +1109,8 @@ static void video_screenshot(struct vo *vo, struct voctrl_screenshot *args)
 
     // Passing an interpolation radius of 0 guarantees that the first frame in
     // the resulting mix is the correct frame for this PTS
-    struct pl_frame *image = (struct pl_frame *) mix.frames[0];
-    struct mp_image *mpi = image->user_data;
+    struct pl_frame image = *(struct pl_frame *) mix.frames[0];
```
Comment (thebombzen) [inline video/out/vo_gpu_next.c:1112]: Are there any stack-size concerns to stack-allocating this struct?

Comment (haasn) [reply video/out/vo_gpu_next.c:1112]: Possibly. Note that we already have a `struct pl_frame` on the stack (`target`). `sizeof(struct pl_frame)` is 616 on my system, currently.

It's also worth pointing out that `pl_render_frame` has stack that's at least `1560` bytes big, too, more in some cases. So we're easily using a few MB of stack space here. Given the limit of 8 MB even on "sane" systems, we might want to try avoiding this.

Comment (haasn) [reply video/out/vo_gpu_next.c:1112]: Err, wait. I was being incredibly stupid and for some reason thinking this was in kilobytes, not bytes.

Let me retract my statement about this being problematic. We're only storing a few *kilobytes* on the stack. Not a few megabytes. I think that alleviates this worry?

Comment (thebombzen) [reply video/out/vo_gpu_next.c:1112]: Yea, I wasn't sure if `struct pl_frame` contained a frame of image data or not. If its size is only in the order of kilobytes that should be no issue.

**Pull request merged**